src/WebAppDIRAC/WebApp/handler/VMDiracHandler.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+
+import ast
+import six
+import json
+
+from DIRAC.Core.Utilities import Time
+from DIRAC.Core.DISET.RPCClient import RPCClient
+from WebAppDIRAC.Lib.WebHandler import WebHandler, asyncGen
+from DIRAC.WorkloadManagementSystem.Client.VMClient import VMClient
+
+
+class VMDiracHandler(WebHandler):
+
+  AUTH_PROPS = "authenticated"
+
+  @asyncGen
+  def web_getInstancesList(self):
+    try:
+      start = int(self.get_argument("start"))
+    except Exception:
+      start = 0
+    try:
+      limit = int(self.get_argument("limit"))
+    except Exception:
+      limit = 0
+
+    sortField = "inst_InstanceID"
+    sortDir = "DESC"
+    if "sort" in self.request.arguments:
+      sortValue = self.get_argument("sort")
+      # converting the string into a dictionary
+      sortValue = ast.literal_eval(sortValue.strip("[]"))
+      sortField = str(sortValue["property"]).replace("_", ".")
+      sortDir = str(sortValue["direction"])
+    sort = [(sortField, sortDir)]
+
+    condDict = {}
+    try:
+      if 'cond' in self.request.arguments:
+        dec = json.loads(self.get_argument("cond"))
+        for k in dec:
+          v = dec[k]
+          if isinstance(v, six.string_types):
+            v = [str(v)]
+          else:
+            v = [str(f) for f in v]
+          condDict[str(k).replace("_", ".")] = v
+    except Exception:
+      raise
```

Comment:
I don't understand the need for `try...except` here...?

Reply:
I have no idea, fixed